src/Symfony/Component/Console/Command/Command.php

```diff
@@ -680,4 +667,34 @@ private function validateName(string $name): void
             throw new InvalidArgumentException(\sprintf('Command name "%s" is invalid.', $name));
         }
     }
+
+    private function getCommandAttribute(?callable $code): ?AsCommand
+    {
+        if (null === $code) {
+            /** @var AsCommand|null $attribute */
+            $attribute = (new \ReflectionClass(static::class)->getAttributes(AsCommand::class)[0] ?? null)?->newInstance();
+
+            return $attribute;
+        }
+
+        $reflection = new \ReflectionFunction($code(...));
+
+        if ($reflection->isAnonymous() || !$class = $reflection->getClosureScopeClass()) {
+            throw new InvalidArgumentException(\sprintf('The command must be an instance of "%s", an invokable object or method.', self::class));
+        }
+
+        /** @var AsCommand|null $attribute */
+        $attribute = ($reflection->getAttributes(AsCommand::class)[0] ?? null)?->newInstance();
+
+        if (!$attribute && $reflection->getName() === '__invoke') {
```

Comment:
Should we reject if the attribute is added to the `__invoke` method? Currently it will behave differently depending if you call
```php
new Command(code: $invokableObject);
// Reads the attribute on the class
```
or
```php
new Command(code: $invokableObject->__invoke(...));
// Reads the attribute on the method first
```